src/main/kotlin/com/chloe/kotlinserv/route/PostGeoDataRoute.kt

```diff
@@ -0,0 +1,53 @@
+package com.chloe.kotlinserv.route
+
+import com.chloe.kotlinserv.http.HttpMethod
+import com.chloe.kotlinserv.http.HttpRequest
+import com.chloe.kotlinserv.http.HttpResponse
+import com.chloe.kotlinserv.http.HttpRoute
+import com.chloe.kotlinserv.model.GeoData
+import com.chloe.kotlinserv.writer.ClickhouseGeoDataWriterImpl
+import com.google.gson.Gson
+import com.google.gson.JsonSyntaxException
+import com.google.inject.Inject
+import com.google.inject.name.Named
+import com.zaxxer.hikari.HikariDataSource
+
+class PostGeoDataRoute @Inject constructor(
+    private val dataSource: HikariDataSource,
+    @Named("geoDataBatchDelay") private val geoDataBatchDelay: Long
+) : HttpRoute {
+    override val endpoint = "/geodata"
+    override val method = HttpMethod.POST
+
+    override val processFunction = { request: HttpRequest ->
+        val batch = ClickhouseGeoDataWriterImpl(ds = dataSource, geoDataBatchDelay = geoDataBatchDelay)
```

Comment:
ClickhouseDataWriter will be created on each new request, lets inject it with Guice